packages/@sanity/core/src/commands/dataset/alias/unlinkAliasHandler.js

```diff
@@ -0,0 +1,59 @@
+import promptForDatasetAliasName from '../../../actions/dataset/alias/datasetAliasNamePrompt'
+import validateDatasetAliasName from '../../../actions/dataset/alias/validateDatasetAliasName'
+import * as aliasClient from './datasetAliasesClient'
+import {ALIAS_PREFIX} from './datasetAliasesClient'
+
+export default async (args, context) => {
+  const {apiClient, output, prompt} = context
+  const [, alias] = args.argsWithoutOptions
+  const client = apiClient()
+
+  const nameError = alias && validateDatasetAliasName(alias)
+  if (nameError) {
+    throw new Error(nameError)
+  }
+
+  const [fetchedAliases] = await Promise.all([aliasClient.listAliases(client)])
+  const aliases = fetchedAliases.map((da) => da.name)
+
+  let aliasName = await (alias || promptForDatasetAliasName(prompt))
+  let aliasOutputName = aliasName
+
+  if (aliasName.startsWith(ALIAS_PREFIX)) {
+    aliasName = aliasName.substring(1)
+  } else {
+    aliasOutputName = `${ALIAS_PREFIX}${aliasName}`
+  }
+
+  if (!aliases.includes(aliasName)) {
+    throw new Error(`Dataset alias "${aliasOutputName}" does not exist `)
+  }
```

Comment:
Maybe instead of mapping the aliases here we could reuse the `linkedAlias` check below? It wasn't obvious to me that the `linkedAlias` would always find the target, given it's in an `if` branch